src/main/java/Duke.java

```diff
@@ -1,10 +1,263 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.util.*;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+    private static final ArrayList<Task> taskArrayList = new ArrayList<>();
+    private static final String PATH_TO_FILE = "data/duke.txt";
+
+    public static void main(String[] args) throws IOException {
+
+        //Always greets user when Duke is run
+        String greeting = "Hello! I'm Duke! \n What can I do for you? \n";
+        System.out.println(greeting);
+        readFromFile(PATH_TO_FILE);
+
+
+        //Use scanner to read input
+        Scanner scanner = new Scanner(System.in);
+
+        String keyWord = scanner.next();
+
+        //6 cases: list, mark, unmark, todo, deadline, event
+        //while loop checks for user exit input("bye")
+        while (!keyWord.equals("bye")) {
+            int taskIndex;
+            Task currentTask;
+            StringBuilder taskDescription;
+            String tempScanner;
+            switch (keyWord) {
+            case "list":
+                System.out.println("\nHere are your tasks:\n");
+                int taskCount = 1;
+                //iterate through taskArray to print out list
+                for (Task t : taskArrayList) {
+                    if (t == null) {
+                        break;
+                    } else {
+                        System.out.println(taskCount + ". " + t + "\n");
+                        taskCount++;
+                    }
+                }
+                if (taskCount == 1) {
+                    System.out.println("You have no task in the list.");
+                }
+                keyWord = scanner.next();
+                break;
+            case "mark":
+                try {
+                    taskIndex = scanner.nextInt() - 1;
+                    currentTask = taskArrayList.get(taskIndex);
+                    currentTask.markAsDone();
+                    System.out.println("\n____________________________________________________________");
```

Comment:
This could be better abstracted to fit OO principles. Perhaps a separate `Parser` / `UI` class?